dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/OrderAndFilterEventChangeLogTest.java

```diff
@@ -268,6 +268,31 @@ void shouldSortChangeLogsWhenOrderingByChangeDescAndChangesOnlyToEventFields()
         () -> assertFieldCreate("geometry", "(-11.419700, 8.103900)", changeLogs.get(4)));
   }
 
+  @Test
+  void shouldSortChangeLogsByNameWhenOrderingByChangeAndDataElementDoesNotHaveFormName()
+      throws ForbiddenException, NotFoundException {
+    EventChangeLogOperationParams params =
+        EventChangeLogOperationParams.builder().orderBy("change", SortDirection.DESC).build();
+    Event event = getEvent("pTzf9KYMk72");
+
+    updateDataValues(event, "DATAEL00001", "value00002", "value00003");
+
+    List<EventChangeLog> changeLogs =
+        eventChangeLogService
+            .getEventChangeLog(UID.of("pTzf9KYMk72"), params, defaultPageParams)
+            .getItems();
+
+    assertNumberOfChanges(7, changeLogs);
+    assertAll(
```

Comment:
Maybe we could have an assertion just to check the order like mapping the changelogs to the order field (if possible) or to the `UID` and make a comment to show what is the value of the name.
Something like:
`expected = List.of("with-option-set", "test-dataleement6", "test-dataelement1", "test-dataelement1", "test-dataelement1", "scheduledAt", "occurredAt")`

This way would be easy to understand why the assertion makes sense.

Reply:
is [this](https://github.com/dhis2/dhis2-core/pull/19729/files#diff-2389d877f149c40712b08260baa60ae93237355a62f970e917258ec0652b1b9dR288) what you meant?